parser/ingester.py

```diff
@@ -69,8 +71,11 @@ def validation_file(self, vf):
                 self._validation_file = vf
             
     @unit.setter
-    def unit(self, unit):
-        if unit:
-            #infer this later (for each set(base element) in the tree, proceed)
-            assert type(unit)==str, "primary unit must be string" 
-        self._unit = unit
+    def unit(self, unit: List) -> None:
+        if type(unit)!=list:
+            if unit:
+                assert type(unit)==str, "units must be string" 
```

Comment:
Can we encapsulate these stringy bois into state vars? Makes it easier and cleaner than having strings in code. Also on line 60, 81...